src/main/scala/mesosphere/marathon/api/akkahttp/v2/QueueController.scala

```diff
@@ -0,0 +1,84 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import java.time.Clock
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{ StatusCodes }
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.api.akkahttp.Directives.{ asLeader, authenticated, get, parameters, pathEndOrSingleSlash, _ }
+import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+import mesosphere.marathon.api.akkahttp.PathMatchers.AppPathIdLike
+import mesosphere.marathon.api.akkahttp.{ Controller, Rejections }
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.core.launchqueue.LaunchQueue
+import mesosphere.marathon.plugin.auth.{ Authenticator, _ }
+import mesosphere.marathon.state.PathId
+
+import scala.async.Async._
+import scala.collection.immutable.Seq
+import scala.concurrent.{ ExecutionContext, Future }
+
+class QueueController(
+    clock: Clock,
+    launchQueue: LaunchQueue)(
+    implicit
+    val actorSystem: ActorSystem,
+    val executionContext: ExecutionContext,
+    val authenticator: Authenticator,
+    val authorizer: Authorizer,
+    val electionService: ElectionService
```

Comment:
I'm not sure we should pass the election service as an implicit. I've done it for the info endpoint and am not happy. What do you think?

Reply:
I agree. Moved